docs/orchestrating-elastic-stack-applications/agent-fleet.asciidoc

```diff
@@ -537,22 +539,147 @@ When deployed with ECK, the Fleet Server Pod makes an HTTP call to itself during
 Solution: enable CNI when starting minikube: `minikube start --cni=true`.
 
 === Storing local state in host path volume
-Elastic Agent when managed by ECK stores local state in a host path volume. This ensures that integrations run by the agent can continue their work without duplicating work that has already been done after the Pod has been recreated for example because of a Pod configuration change.
-Multiple replicas of an agent, for example Fleet Servers, can not be deployed on the same underlying Kubernetes node as they would try to use the same host path. If local state storage in host path volumes is not desired this can be turned off by configuring an `emptyDir` volume instead:
+Elastic Agent managed by ECK stores local state in a host path volume by default. This ensures that integrations run by the agent can continue their work without duplicating work that has already been done after the Pod has been recreated for example because of a Pod configuration change. Multiple replicas of an agent, for example Fleet Servers, can not be deployed on the same underlying Kubernetes node as they would try to use the same host path. There are 2 options for managing this feature:
+
+1. If local state storage in `hostPath` volumes is not desired this can be turned off by configuring an `emptyDir` volume instead.
+2. If local state storage is still desired but running the Agent container as root is not allowed, then you can run a `DaemonSet` that adjusts the permissions for the Agent local state on each Node prior to running Elastic Agent. *Note that this `DaemonSet` must be `runAsUser: 0` and possibly `privileged: true`*. *Also note the Kibana changes required to trust the Elasticsearch CA*.
 
 [source,yaml]
 ----
+# Using an emptyDir instead of a hostPath volume.
 apiVersion: agent.k8s.elastic.co/v1alpha1
 kind: Agent
 metadata:
   name: fleet-server
 spec:
   deployment:
-    replicas: 3
     podTemplate:
       spec:
+        securityContext:
+          # Gid 1000 is the default group at which the Agent container runs.
+          # Adjust as necessary if `runAsGroup` has been modified.
+          fsGroup: 1000
```

Comment:
Why not use callouts here? Comments are rendered very lightly in the current Elastic style sheet.